### ISSUE TITLE
bug: SNS publishing from step functions seems to not construct the Message attribute correctly

### ISSUE THREAD

**Issue by silenceisgolden** [localstack/localstack#10364]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Right now, I have a state machine definition that contains:

```
ScanForRecords: {
        Type: 'Task',
	Resource: 'arn:aws:states:::sns:publish.waitForTaskToken',
		Parameters: {
			TopicArn: '<valid_arn_redacted>',
			Message: {
			        'Input.$': '$',
				'TaskToken.$': '$$.Task.Token',
			},
		},
	Next: 'EvaluateScanContinue',
},
```

When this step function runs in localstack and delivers the message to the SNS topic, the Message parameter is just

```
{
  ... other attributes
  "Message": "Input"
  ... more attributes
}
```

### Expected Behavior

If the input I provide to the step function execution is

```
{
  "a":true
}
```

and a task token is generated, I would expect the Message received to be very similar to the following:

```
{
  ... other attributes
  "Message": "{\"Input\":{\"a\":true},\"TaskToken\":\"tokenvalue\"}"
  ... more attributes
}
```

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

* Create an SNS topic that delivers to a queue, lambda, something where you can look at the result    
* Create a state machine with one State that is similar to the above example and use the arn from the topic for the "TopicArn" value
* Create a new execution in that state machine with a stringified JSON input
* Observe that the SNS message that is delivered just has a simple string instead of the JSON stringified payload that includes the task token

### Environment

```markdown
- OS: MacOS 14.3.1 (Docker Desktop 4.27.2, Docker Engine 25.0.3)
- LocalStack: 3.1.0
```

### Anything else?

I have not encountered this issue when using the amazon/aws-stepfunctions-local docker container. I was migrating our local environment to use Map state locally and since amazon/aws-stepfunctions-local does not support that, I tried to use the localstack emulator instead. We have other state machines running locally that use this parameter interpolation pattern that do not need Map state that seem to run just fine with amazon/aws-stepfunctions-local.

**Comment by MEPalma** [comment]
Hello @silenceisgolden, many thanks for compiling this report. It seems that the problem has been addressed in subsequent releases of LocalStack. Could you please verify if it persists in the latest version?

**Comment by silenceisgolden** [comment]
Confirmed fixed in 3.2.0! Thanks for the heads up @MEPalma
